src/app/dashboard/competitions/[competitionId]/register-form/create/components/CreateCompetitionRegisterForm.tsx

```diff
@@ -0,0 +1,217 @@
+    }
+  }
+
+  function handleDeleteField(index: number) {
+    setNewFieldName(newFieldName.filter((_, i) => i !== index));
+    setNewFieldType(newFieldType.filter((_, i) => i !== index));
+  }
+
+  return (
+    <div className="card bg-base-100 shadow-xl">
+      <div className="mt-4">
+        <GoBackButton />
+      </div>
+      <div className="card-body">
+        <h1 className="card-title mb-6 text-2xl">Crear formulario de registro para {competition.name}</h1>
+
+        <form id={form.id} onSubmit={form.onSubmit} action={formAction} className="space-y-4">
+          <FormInput
+            label="Título"
+            name={fields.title.name}
+            type="text"
+            required
+            placeholder="Título del formulario de registro"
+            errors={fields.title.errors}
+          />
+
+          <FormInput
+            label="Descripción"
+            name={fields.description.name}
+            type="text"
+            required
+            placeholder="Descripción del formulario de registro"
+            errors={fields.description.errors}
+          />
+
+          <FormCheckbox
+            label="Gratuito"
+            name={fields.isFree.name}
+            checked={isFreeChecked}
+            onChange={() => setIsFreeChecked(!isFreeChecked)}
+            errors={fields.isFree.errors}
+          />
+
+          {isFreeChecked === false && (
+            <>
+              <PriceInput
+                name={fields.price.name}
+                errors={fields.price.errors}
+              />  
+              <br></br>
+              <Toogle
+                label={isPaymentToggleChecked ? "Datos de pago" : "Link de pago"}
+                checked={isPaymentToggleChecked}
+                onChange={() => setIsPaymentToggleChecked(!isPaymentToggleChecked)}
+              />
+
+              <input type="hidden" name={fields.isPaymentToggleChecked.name} value={isPaymentToggleChecked ? 'true' : 'false'} />
+              {isPaymentToggleChecked ? (
+                <>
+                  <FileInput
+                    label="Datos de pago"
+                    name={fields.paymentData.name}
+                    accept=".pdf, .jpg, .jpeg, .png, .svg"
+                    errors={fields.paymentData.errors}
+                  />
+                </>
+              ) : (
+                <>
+                  <FormInput
+                    label="Link de pago"
+                    name={fields.paymentLink.name}
+                    type="text"
+                    errors={fields.paymentLink.errors}
+                  />
+                  <p className="mt-8 text-sm text-gray-500">Recomendamos el uso de <a href="https://www.slach.cl/accounts" target="_blank" rel="noopener noreferrer" className="text-blue-500 hover:text-blue-600">Slach </a>  
+              para recibir pagos. Puedes crear un link de pago y pegarlo en el siguiente campo o adjuntar tus datos de pago como pdf o imagen.</p>
+                </>
```

Comment:
Ya que estamos en la opción de Link de pago, no mostraría información relacionada a adjuntar datos de pago.

```suggestion
                  <p className="mt-8 text-sm text-gray-500">Recomendamos el uso de <a href="https://www.slach.cl/accounts" target="_blank" rel="noopener noreferrer" className="text-blue-500 hover:text-blue-600">Slach </a>  
              para recibir pagos. Puedes crear un link de pago y pegarlo en el siguiente campo.</p>
                </>
```